### PR TITLE
Implement _sys_memmove, minor fixes

### LINKED ISSUE #1468: Super Puzzle Fighter II Turbo HD Remix [NPUB90041]  invalid map<K, T> key error
The game doesn´t launch and shows an error message since this branch: v0.0.0.6-433-gb77666c

https://github.com/Bigpet/rpcs3/releases/download/v0.0.0.6-433-gb77666c/rpcs3-v0.0.0.6-433-gb77666c-win_x64.zip
https://github.com/Bigpet/rpcs3/releases/tag/v0.0.0.6-433-gb77666c

The error message: 
```
{PPU Thread[0x1] (main_thread)[0x002f2178]} cellGame: cellHddGameCheck(version=0, dirName=*0x10009d88, errDialog=0, funcStat=*0x306728, container=13)
{PPU Thread[0x1] (main_thread)[0x002f2178]} Exception: invalid map<K, T> key
Please report this to the developers.
```

Other informations that could help:

-The game in rpcs3 gamelist shows the App. Version as: unknown

-After it I did a test: I deleted the param.sfo of the Super Puzzle Figther II Turbo HD Remix game folder [of course the game doesn´t appear anymore in the rpcs3 menu gamelist]

-So I replaced the file with the Sonic CD param.sfo:
The game shows in the rpcs3 menu gamelist with the Sonic CD name but the Super Puzzle Fighter II Turbo HD Remix icon and when you click it launch the game and it runs perfectly as in the previous versions.

So i think it´s something related to the param.sfo file. 

super puzzle
super puzzle2

### PR DESCRIPTION
**Pull request by tambry**
~~cellHddGameCheck function definitely doesn't set getParam values.~~
Also implemented _sys_memmove for Resogun.

### CODE REVIEW

**Comment by MyaniPT** [comment]
@tambry Do you have any links of the master build with this added in it? I'd like to test and I don't know how to build it myself as for the moment

**Comment by PZerua** [comment]
@MyaniPT You can use the AppVeyor build clicking in "Details" below and then in Artifacts. Here is the direct link: https://ci.appveyor.com/project/rpcs3/rpcs3/build/1.0.1484/artifacts

**Comment by MyaniPT** [comment]
Thanks @PZerua 
This fixed the issues with #1468 , at least for me in the games I tested

**Comment by tambry** [comment]
@MyaniPT Just grab one from AppVeyor artifacts. But remember to not post any results with the build on the forum or open any issues while using it.
If there are any regressions let me know.

File: rpcs3/Emu/SysCalls/Modules/sys_libc.cpp [hunk 1]
```diff
@@ -191,9 +191,13 @@ s32 _sys_memchr()
 	throw EXCEPTION("");
 }
 
-s32 _sys_memmove()
+vm::ptr<void> _sys_memmove(vm::ptr<void> dst, vm::cptr<void> src, size_t size)
```
Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/Modules/sys_libc.cpp]: u32 size

File: rpcs3/Emu/SysCalls/Modules/sys_libc.cpp [hunk 2]
```diff
@@ -191,9 +191,13 @@ s32 _sys_memchr()
...
 {
-	throw EXCEPTION("");
+	sysPrxForUser.warning("_sys_memmove(s1=*0x%x, s2=*0x%x, size=%d)", dst, src, size);
```
Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/Modules/sys_libc.cpp]: Logged as s1 and s2 for some reason

File: rpcs3/Emu/SysCalls/Modules/cellGame.cpp [hunk 3]
```diff
@@ -75,35 +76,16 @@ s32 cellHddGameCheck(PPUThread& ppu, u32 version, vm::cptr<char> dirName, u32 er
...
-		strcpy_trunc(get->getParam.dataVersion, psf.at("APP_VER").as_string());
-		strcpy_trunc(get->getParam.titleId, psf.at("TITLE_ID").as_string());
-
-		for (u32 i = 0; i < CELL_HDDGAME_SYSP_LANGUAGE_NUM; i++)
-		{
-			strcpy_trunc(get->getParam.titleLang[i], psf::get_string(psf, fmt::format("TITLE_%02d", i)));
-		}
-	}
-
-	// TODO ?
```
Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/Modules/cellGame.cpp]: What's there?

Comment (tambry) [inline rpcs3/Emu/SysCalls/Modules/cellGame.cpp]: I'm not sure. There really shouldn't be anything, but this function return results need to be confirmed on a PS3 probably. getParam is definitely not set by this function, though.

**Comment by vlj** [comment]
Need rebase

File: rpcs3/Emu/SysCalls/Modules/cellGame.cpp (hunk 3 above)
Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/Modules/cellGame.cpp]: @tambry `get` variable is exactly what has to be set before it's passed to the callback.

**Comment by Nekotekina** [comment]
This fix will likely break something. @tambry What was exactly the problem, exception in cellHddGameCheck? PSF properties could be read with default value.

**Comment by tambry** [comment]
@Nekotekina You're right. After further reading the documentation you're right. I have currently removed the commit, as I think the correct behaviour should be confirmed on a real PS3, but I currently don't have the time to do that.

**Comment by elfogames** [comment]
Without "cellHddGameCheck function definitely doesn't set getParam values" Fix
Super Street Fighter 2 Turbo HD  isn't loadable anymore :( and this error is back:

regression

**Comment by MyaniPT** [comment]
Yeah, there were a bunch of games that don't load due to that error, and the previous (incorrect) commit you made fixed it. The error is back now unfortunately

**Comment by tambry** [comment]
I'll try to write a test for checking the behaviour next week or maybe even on Friday.

**Comment by vlj** [comment]
need rebase so I can merge it

**Comment by tambry** [comment]
@vlj I'm aware of this. I'm currently also looking into a couple bugs in the recently merged cellPngDec rewrite. I'd like to also get them into this PR.

On a bit of off-topic, there's a Khronos Vulkan webinar starting in ~25 minutes. You might want to check it out, since you're working on the Vulkan backend.
Link for registration: https://www.khronos.org/news/events/vulkan-webinar

**Comment by tambry** [comment]
Rebased and have added some HLE fixes along with a commit to improve/clarify the general exception messages.  
The exception messages were too broad and only have brought reports about already known problems. Anything that would be thrown would simply be reported. This causes a lot of new users to simply report these "problems" and has caused quite a lot of spam on the forums over the past month.  
This PR probably needs to be re-reviewed.

**Comment by Nekotekina** [comment]
Changing the error message won't change the amount of idiots. If you really want to improve it, it'll require some different kind of work. For example, let's say there is ILLEGAL INSTRUCTION error code. We could handle this error specifically and tell the user that he is really mean trying to run rpcs3 on really OLD CPU which doesn't even support SSSE3. Or something.

**Comment by Ekaseo** [comment]
the text in pac-man is unreadable for some reason, its fixed when you turn png dec LLE module on

**Comment by tambry** [comment]
@Ekaseo Could probably open an issue for it after this is merged. I could probably eventually look into.

Might be noteworthy, that with cellPngDec LLE, there are some internal errors in cellPngDec, which causes it to fail to decode certain pictures, thus the main menu is missing a lot of effects. (at least for me)

@Nekotekina I have added specific exception related error messages, as you suggested. I have also left a generic message in for unknown exceptions.

File: Utilities/Thread.cpp [hunk 4]
```diff
@@ -22,10 +22,9 @@
 static void report_fatal_error(const std::string& msg)
 {
 #ifdef _WIN32
-	const auto& text = msg + "\n\nPlease report this error to the developers. Press (Ctrl+C) to copy this message.";
-	MessageBoxA(0, text.c_str(), "Fatal error", MB_ICONERROR); // TODO: unicode message
+	MessageBoxA(0, msg.c_str(), "Fatal error", MB_ICONERROR); // TODO: unicode message
 #else
-	std::printf("Fatal error: %s\nPlease report this error to the developers.\n", msg.c_str());
+	std::printf(msg.c_str());
```
Comment (Nekotekina) [inline Utilities/Thread.cpp]: Wtf. First arg must be a constant string.

Comment (tambry) [inline Utilities/Thread.cpp]: No? http://puu.sh/nezBB/09df20e430.png

Comment (paulsapps) [inline Utilities/Thread.cpp]: "test" is const char* and msg.c_str() is also const char*

Comment (Nekotekina) [inline Utilities/Thread.cpp]: @tambry Great, now put %s into msg and see it crash.

Comment (tambry) [inline Utilities/Thread.cpp]: [Doesn't crash though](http://puu.sh/neR9D/734b5a1709.png). But I see your point.

File: Utilities/Thread.cpp [hunk 5]
```diff
@@ -22,10 +22,9 @@
 static void report_fatal_error(const std::string& msg)
 {
 #ifdef _WIN32
-	const auto& text = msg + "\n\nPlease report this error to the developers. Press (Ctrl+C) to copy this message.";
-	MessageBoxA(0, text.c_str(), "Fatal error", MB_ICONERROR); // TODO: unicode message
+	MessageBoxA(0, msg.c_str(), "Fatal error", MB_ICONERROR); // TODO: unicode message
 #else
-	std::printf("Fatal error: %s\nPlease report this error to the developers.\n", msg.c_str());
+	std::printf("%s", msg.c_str());
```
Comment (Nekotekina) [inline Utilities/Thread.cpp]: And "Fatal error", and "\n" got removed... Do you ever look at the diff, what you are actually doing?

**Comment by tambry** [comment]
@Nekotekina Anything else or is it OK?

**Comment by Nekotekina** [comment]
Without the hint about Ctrl+C, people will possibly upload 100 MB screenshots badly cropped in paint.

**Comment by tambry** [comment]
@Nekotekina Added a hint about CTRL+C for unhandled exceptions.

**Pull request merged**